test/test-tcp-open.c

```diff
@@ -180,3 +191,30 @@ TEST_IMPL(tcp_open) {
   MAKE_VALGRIND_HAPPY();
   return 0;
 }
+
+
+TEST_IMPL(tcp_open_twice) {
+  uv_tcp_t client;
+  uv_os_sock_t sock1, sock2;
+  int r;
+
+  startup();
+  sock1 = create_tcp_socket();
+  sock2 = create_tcp_socket();
+
+  r = uv_tcp_init(uv_default_loop(), &client);
+  ASSERT(r == 0);
+
+  r = uv_tcp_open(&client, sock1);
+  ASSERT(r == 0);
```

Comment:
Suggestion: add another test asserting that calling `uv_tcp_open(&client, sock1)` again fails with UV_EBUSY.